thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -253,7 +255,23 @@ public Builder registerAction(
                 @NonNull String alias,
                 @NonNull String actionName,
                 @NonNull Class<? extends Action> actionClass){
-            //TODO: validate actionClass, should contain only 1 field
+            // validate field of actionClass
+            List<Field> fields = new ArrayList<>();
+            for (Field field : actionClass.getDeclaredFields()) {
+                if (!field.getName().equals("this$0")){ // in case of inner class, has this field
```

Comment:
> Can we create instances non static inner class with Gson?

I can not understand well.  

here is an example to use inner class for action. 

```java
class ThingIFAPIConsumer {
   ThingIFAPI api;
   
   class TurnPower implements Action {...}

   public init() {
       // init api, register action here.
   }

   public postCommand(...) {
      // use api to post command
   }
}
```